teerex/src/lib.rs

```
@@ -148,9 +149,13 @@ decl_module! {
 
 		/// The integritee worker calls this function for every processed parentchain_block to confirm a state update.
 		#[weight = (<T as Config>::WeightInfo::confirm_processed_parentchain_block(), DispatchClass::Normal, Pays::Yes)]
-		pub fn confirm_processed_parentchain_block(origin, block_hash: H256, trusted_calls_merkle_root: H256) -> DispatchResult {
-			let sender = ensure_signed(origin)?;
+		pub fn confirm_processed_parentchain_block(origin, block_hash: T::Hash, trusted_calls_merkle_root: H256) -> DispatchResult {
+			let sender = ensure_signed(origin.clone())?;
 			Self::is_registered_enclave(&sender)?;
+			if system::pallet::BlockHash::<T>::iter_values().find(|bh| *bh == block_hash).is_none() {
+				Self::unregister_enclave(origin)?;
+				return Err(<Error<T>>::UnknownBlockHash.into());
```

Not sure we should return an error here. I think it's what make benchmarking fail.
There was no error in the code execution. It fulfilled it's purpose: get rid of corrupted worker.